lib/ui/pages/chat_page.dart

```diff
@@ -136,41 +111,7 @@ class _ChatPageState extends State<ChatPage> {
     });
   }
 
-  void _copyToClipboard(String text) {
-    Clipboard.setData(ClipboardData(text: text));
-  }
-
-  void _regenerateResponse(List<ChatMessage> messages) async {
-    if (_isLoading || messages.isEmpty) return;
-
-    if (!messages.last.isUser) {
-      setState(() => messages.removeLast());
-    }
-
-    setState(() => _isLoading = true);
-
-    try {
-      final botMessage = await _repository.getNextMessage(_messages);
-
-      setState(() {
-        messages.add(botMessage);
-        _isLoading = false;
-      });
-
-      _scrollToBottom();
-    } catch (e) {
-      setState(() => _isLoading = false);
-      _showError(e.toString());
-    }
-  }
-
-  void _showError(String msg) {
-    ScaffoldMessenger.of(context).showSnackBar(
-      SnackBar(content: Text(msg), backgroundColor: AppColors.alertRed),
-    );
-  }
-
-  Widget _buildMessageBubble(ChatMessage message) {
+  Widget _buildMessageBubble(ChatMessage message, ChatCubit cubit) {
```

Comment:
create separate common widget (probably we will reuse it) and make it's behaviour independent from ChatCubit to give us ability for reuse easy